### ISSUE TITLE
[java] UseLocaleWithCaseConversions false positive

### ISSUE THREAD

**Issue by vanniktech** [pmd/pmd#356]
**Rule Set:**

```xml
<rule ref="rulesets/java/design.xml"/>
```

**Description:**

**Code Sample demonstrating the issue:**

```java
public class Test {
  public void test() {
    new Test().toUpperCase();
  }

  public void toUpperCase() {
  }
}
```

**Running PMD 5.5.5 through:** *Gradle*

`new Test().toUpperCase();` will be flagged by PMD:

```xml
<violation beginline="5" endline="5" begincolumn="5" endcolumn="28" rule="UseLocaleWithCaseConversions" ruleset="Design" package="net.familo.android.image.loader.strategies" class="Test" method="test" externalInfoUrl="https://pmd.github.io/pmd-5.5.5/pmd-java/rules/java/design.html#UseLocaleWithCaseConversions" priority="3">
When doing a String.toLowerCase()/toUpperCase() call, use a Locale
</violation>
```

However there's no method for passing in a Locale so this line should not be flagged. Also Test is not of type String so the message is a bit misleading too.

**Comment by jsotuyod** [comment]
@vanniktech thanks for the report, certainly a bug.

This rule requires [full type resolution](https://github.com/pmd/pmd/wiki/Project-Ideas-%5BMature%5D#complete-type-resolution-for-java) to filter out these false positives, hopefully that project will be completed over the next few months.

**Comment by vanniktech** [comment]
Can this be tackled with the current type resolution that exists?

**Comment by jsotuyod** [comment]
@vanniktech for most scenarios, such as the one in the example, the current state of type resolution is probably enough, yes. More complex scenarios relaying on type inference will still not work (that part is still WIP), but the changes to the rule to make it work are the same.

**Comment by vanniktech** [comment]
No longer interested in this.
